### PR TITLE
106 cart bug - COMPLETED-- no longer W-I-P

### LINKED ISSUE #106: Cart bug: Splitting payments is not correctly setting the amount owed.
The purchase flow looks like this:

1. User purchases two supporting memberships, and pays fully, using the cart
2. User attempts to upgrade the memberships to Attending in one case, First Worldcon in another

The state now shows an incorrectly fully paid membership on all counts, despite not having had payments made to them.

### PR DESCRIPTION
**Pull request by Lasiorhine**
Work in progress for the fee-splitting cart bug

### CODE REVIEW

**Comment by offbyone** [comment]
I think this'll also need some documentation covering how payments are tracked now. Does this affect standard purchases, or what? Basically, everything I'm going to ask you about in an hour or so... let's write down :)

File: app/controllers/charges_controller.rb [hunk 1]
```diff
@@ -28,19 +28,15 @@ def new
...
     price_steps = PaymentAmountOptions.new(@outstanding_amount).amounts
-
     @price_options = price_steps.reverse.map do |price|
       [price.format, price.cents]
     end
   end
 
   def create
     charge_amount = Money.new(params[:amount].to_i)
-
```
Comment (offbyone) [inline app/controllers/charges_controller.rb:41]: nit: I actually prefer whitespace for readability.

**Comment by Lasiorhine** [comment]
Pending completion of the CI checks, this is officially no longer a work in progress.

**Comment by Lasiorhine** [comment]
> I think this'll also need some documentation covering how payments are tracked now. Does this affect standard purchases, or what? Basically, everything I'm going to ask you about in an hour or so... let's write down :)

I added some comments to SetMembership, where this comes into play.  LMK if we need more than that.  Maybe we need something like a whitepaper somewhere?

**Review by offbyone — APPROVED** [review]

**Comment by offbyone** [comment]
This'll need to be rebased on top of staging.

**Pull request merged**